src/django_components/dependencies.py

```diff
@@ -650,9 +663,20 @@ def _get_script_tag(
     script = get_script_content(script_type, comp_cls)
 
     if script_type == "js":
-        return f"<script>{_escape_js(script)}</script>"
+        if "</script" in script:
```

Comment:
I think this is still too brittle. A starting <script> tag would break this too (they can't test so the parser implicitly closes the previous one). I think you can put non-printable characters in "</script" to still have it execute in some browsers. Other browser tags may close this as well, like </html>.

Reply:
Hm, could you point to some examples/reference? My own testing in Chrome as well as ChatGPT suggest that:
1. Inserting `<script>` inside `<script>` does NOT split it original script
2. Any other end tag other than `</script>` is ignored inside `<script>`

Reply:
I'm testing in codepen in Firefox (double-verified in Edge):
* <script>document.write("<script>HELLO")</script> renders nothing, because the <script> tags ends the first script after the first quote
* </script> containing control chars doesn't seem to work.

Going back to the original problem here: Are we trying to prevent people from making accidental mistakes, or is this a security feature?

Reply:
> Going back to the original problem here: Are we trying to prevent people from making accidental mistakes, or is this a security feature?

Preventing people from accidental mistakes. In one of the future PRs, I want to add component attributess similar to `template_name`, which would specify `Component.js/css` as a separate file. But although written in a separate file, they would eventually be inserted into the HTML as `<script>` and `<style>` tags.

And I imagine it could be fairly hard-to-find bug, if I accidentally broke the JS script because I included `</script>` within a string or a comment.

---

Hm, I still can't reproduce the errors.

Yes, your example with

```
<script>document.write("<script>HELLO")</script>
```

doesn't show anything in the DOM, but that's not because of what we're discussing here. It's because you're introducing an un-terminated `<script>` tag into the DOM. And what the browser does is that it looks for the closest `</script>`. Which, if there is none, means that all the rest of the page is interpreted as contents of the `<script>` tag.

See the highlighted section in the screenshot below. The section contains `</body>`, `</html>`, and a comment, which were part of the original document, but are now treated as contents of the inserted `<script>`. The browser then reaches the end of the document without terminating the `<script>`, nor `<body>` and `<html>`, so it adds the terminating `</script>`, `</body>` and `</html>` tags, which you can see below the highlighted section.

<img width="508" alt="Screenshot 2024-12-19 at 08 16 41" src="https://github.com/user-attachments/assets/ba59909f-0e57-4b06-b974-57af1226aa2b" />

---

On the other hand, I got consistently the same results on:
- Chrome v118 (MacOS 14.5)
- Chrome v131 (Windows 11)
- Firefox v109  (Mac)
- Firefox v133  (Win)
- Safari v17.5 (Mac)
- Edge v131 (Win)

The consistent behaviour is that:
1. When I insert something into the `</script>` tag (e.g. `</\u200Bscript>`), the browser does NOT treat is as proper end tag, and instead only as a string 
2. If I insert extra `<script>` inside another script tag, nothing happens, browser treats it as a text
3. If I insert extra `</script>` inside another script tag, the script ends prematurely

Of the 3, behavior 1 and 2 is expected and desirable.  It's only behaviour 3 that I want to guard against. And it seems that checking only for the presence of `</script>` is sufficient.

Reply:
Thank you @JuroOravec, you have me convinced! OK to merge from me! :)